source/common/http/conn_manager_impl.cc

```diff
@@ -1594,23 +1607,26 @@ void ConnectionManagerImpl::ActiveStream::sendLocalReply(
         Utility::EncodeFunctions{
             [&](ResponseHeaderMap& response_headers, Code& code, std::string& body,
                 absl::string_view& content_type) -> void {
-              connection_manager_.config_.localReply().rewrite(
-                  request_headers_.get(), response_headers, stream_info_, code, body, content_type);
+              connection_manager_.config_.localReply().rewrite(filter_manager_.requestHeaders(),
+                                                               response_headers, stream_info_, code,
+                                                               body, content_type);
             },
             [&](ResponseHeaderMapPtr&& response_headers, bool end_stream) -> void {
               if (modify_headers != nullptr) {
                 modify_headers(*response_headers);
               }
-              response_headers_ = std::move(response_headers);
-              encodeHeaders(*response_headers_, end_stream);
+              // TODO(snowp): This is kinda awkward but we need to do this so that the access log
+              // sees these headers. Is there a better way?
```

Comment:
Shouldn't sendLocalReply just always get routed through the filter manager?

Reply:
The comment above explains why we skip the filters here: 
```
    // In this case, at least the header and possibly the body has started
    // processing through the filter chain, but no non-informational headers
    // have been sent downstream. To ensure that filters don't get their
    // state machine screwed up, bypass the filter chain and send the local
    // reply directly to the codec.
```

Reply:
I see, OK fair enough to leave this TODO for now.